### DISCUSSION TITLE
List of good ideas

### DISCUSSION THREAD

**Discussion by stmcculloch** [stmcculloch/arc-overhang#14] [Ideas]
There are 5 areas where arc overhangs need to be improved: 
1. Prevent warping
2. Increase speed
3. Improve surface quality
4. Improve reliability
5. Optimize code

If you'd like to start working on any of these, feel free to create an issue and document progress!
If you have any other ideas that aren't on this list please leave them below as a comment.

## 1. Prevent warping

**Tested:**
- Add support pillars underneath arc centers and around perimeters to anchor the edges. In progress by @rvmn. [See here](https://github.com/rvmn/SuperPleccer/discussions/4)

**Todo:** 
- Use tree supports instead of custom support pillars to further minimize support waste. [More info here](https://github.com/stmcculloch/arc-overhang/issues/13)
- Print second & third layer slower than normal
- Repeat the arc overhang pattern again on the next 2-3 layers to increase stiffness before going back to default slicer settings.
- Use arcs as a "support floor" that traditional supports get printed on top of. Would save lots of material for tall parts.
- Maximum overhang area -  Only use arc overhangs for small overhanging areas and normal support for large overhangs. 

## 2. Increase speed

**Tested:** 
- Print small arcs slowly, but increase speed for larger arcs. Cooling is the main limitation. [Seems to help slightly](https://youtu.be/XSqvxzFnL9g?t=113)

**Todo:** 
- Eliminate small arcs entirely using [flood-fill](https://github.com/stmcculloch/arc-overhang/issues/12)
- Try on printers with better part-cooling including aux-fan. E.g. Bambulabs X1 carbon.

## 3. Improve surface quality

**Tested:**
- [Use the same center as long as possible](https://github.com/stmcculloch/arc-overhang/issues/10#issue-1497025939) Initially suggested by @MULTIPL3X
- Eliminate tiny arcs with [flood-fill](https://github.com/stmcculloch/arc-overhang/issues/12) Tested by @ServAlex. Works very well. Similar to above but algorithm is slightly different and uses an expanding buffer starting from the original starting line instead of a circular arc.
- Breadth-first instead of depth-first. (Added by Nicolai. [Works well, definitely the way to go for anyone trying recursive generation](https://github.com/nicolai-wachenschwan/arc-overhang-prusaslicer-integration))
- Offset arc centers inward to hide the smallest arcs. [Better in theory than in practice](https://youtu.be/XSqvxzFnL9g?t=89)
- Combine arcs + add more perimeters. [Big improvement here](https://youtu.be/XSqvxzFnL9g?t=127)
- Gradual interpolation to the outer walls using variable line width. Started by @ServAlex. [See here](https://github.com/stmcculloch/arc-overhang/issues/11)

**Todo:** 
- Other space filling patterns like [Koch curves or Hilbert curves](https://github.com/stmcculloch/arc-overhang/issues/6)

## 4. Reliability:

**Tested:** 
- Handle holes
  - Original arc overhangs work for this using breadth-first search. Added by @nicolai-wachenschwan. [Looks promising](https://github.com/nicolai-wachenschwan/arc-overhang-prusaslicer-integration)
 
  
**Todo:**
- [Wave pattern](https://github.com/stmcculloch/arc-overhang/issues/10#issuecomment-1374636774) 
- Hole perimeters need to be printed slowly + with overlap so they don't delaminate inward. Must print in order from inside to outside.
- Z-hop enabled specifically for these (in case of warping, prevents hitting curled edges)

## 5. Optimize code

**Todo:** 
- Hexagons instead of arcs (fewer lines of gcode)
- Minimum line length (avoids very small gcode moves)

**Comment by rvmn** [comment]
Regarding 4. Reliability. in particular the wave pattern idea: [update coming soon](https://github.com/rvmn/SuperPleccer/discussions/10)
